provider/aws/processes.go

```diff
@@ -308,6 +312,10 @@ func (p *AWSProvider) taskProcesses(tasks []string) (structs.Processes, error) {
 
 	pss := structs.Processes{}
 
+	// THROTTLING: taskProcesses
+	// This one looks hairy, but if we could find some way
+	// to consolidate into a single API call rather than
+	// one for each task, we might get some gains here.
```

Comment:
My comment here's a little off on further reflection. Specifically I think the loop with `p.fetchProcess` below this is what's generating a bunch of calls